virtualization-and-containers/material/exercise_vm_containers_text.md

```diff
@@ -0,0 +1,215 @@
+  ```
+
+  You might want to add this line to your `.bashrc` to make this change persistent. Note, that Vagrant will still create a hidden `.vagrant` directory in the your working directory. However, the `.vagrant` directory normally does not need much space on your hard drive.
+- [Vagrant Homepage](https://www.vagrantup.com/)
+- [Vagrant Introduction](https://www.vagrantup.com/intro)
+- [VirtualBox Manual](https://www.virtualbox.org/manual/UserManual.html)
+
+## Containers Using Docker
+
+Similar to the previous task we want to set up a Docker container for testing. It will contain similar functionality as the Vagrant box.
+
+### Tasks (Docker)
+
+- Fork the [TODO: "Exercise Containers"](https://gitlab-sim.informatik.uni-stuttgart.de/sse-test-group/exercise-containers) repository and create a branch to work on the fork. It will contain a `README.md` and a file called `Dockerfile`.
+- Currently, the [`Dockerfile`](https://docs.docker.com/engine/reference/builder/) is empty. Edit it so an image with the following properties is build:
+  - Your virtual machine should be based on the [`ubuntu:20.04` image](https://hub.docker.com/_/ubuntu).
+  - Create a new file with the name `testfile` and add it to the repository. The file should contain your GitLab username as text. Add the file to your container and place it in directory `/testfiles`.
+  - Set the environment variable `ENV_TEST_VARIABLE` to have the value of your GitLab username.
+  - Make sure that the default command to be executed when running the container is `/bin/bash`.
+  - Install `neofetch`.
+  - **Note:** In the `Dockerfile` you do not have to prefix commands such as `apt` with `sudo` since the script is executed with superuser rights.
+- Test your Docker container locally. Are all variables set and files available?
+  - **Note:** If you rebuild your container often, you might end up with dangling containers. You can remove them (and unused images, containers,or objects) with [`docker [image|container|system] prune`)](https://docs.docker.com/engine/reference/commandline/system_prune/) depending on what you want to remove.
+- Open a merge request in the GitHub Repository [TODO: "Exercise Containers"](https://gitlab-sim.informatik.uni-stuttgart.de/sse-test-group/exercise-containers).
+  - As title choose "[`USERNAME`] Docker Container Recipe"
+  - Make sure all files are up to date (`testfile`, `Dockerfile`)
+  - Add the label `Docker` label to the issue and assign the issue to the lecturers `jaustar` and `uekermbn`.
+  - Double-check that all files are in the repository and up to date. Check everything carefully since a CI pipeline will try to build your Docker recipe.
+  - If everything looks good, create the issue.
+- Check whether the CI pipeline could verify your submission. This may take some time due to the number of concurrent submission being limited. If read the error message carefully and fix your submission by creating a new commit.
```

Comment:
```suggestion
- Check whether the CI pipeline could verify your submission. This may take some time due to the number of concurrent submission being limited. If it fails, read the error message carefully and fix your submission by creating a new commit.
```